### PR TITLE
update the translation of `Object.tolocalestring()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

update the translation of `Object.tolocalestring()`
### Motivation

update the translation of `Object.tolocalestring()`
### Additional details

none

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 1]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
 slug: Web/JavaScript/Reference/Global_Objects/Object/toLocaleString
 ---
 
-{{JSRef("Global_Objects", "Object")}}
+{{JSRef}}
 
-**`toLocaleString()`** 方法返回一个该对象的字符串表示。此方法被用于派生对象为了特定语言环境的目的（locale-specific purposes）而重载使用。
+**`toLocaleString()`** 方法返回一个表示对象的字符串。该方法旨在由派生对象重载使用，以达到特定于语言环境的目的。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:8]:
```suggestion
**`toLocaleString()`** 方法返回一个表示对象的字符串。该方法旨在由派生对象重写，以达到其特定于语言环境的目的。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 2]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
 
-```plain
-obj.toLocaleString();
+```js-nolint
+toLocaleString()
 ```
 
+### 参数
+
+没有参数。但是，重载此方法的所有对象最多只能接受两个参数，分别对应于 `locales` 和 `options`，例如 {{jsxref("Date.prototype.toLocaleString")}}。参数位置不用于任何其他目的。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:20]:
```suggestion
没有参数。但是，重写此方法的所有对象最多只能接受两个参数，分别对应于 `locales` 和 `options`，例如 {{jsxref("Date.prototype.toLocaleString")}}。这些参数位置不应该用于任何其他目的。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 3]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+
 ### 返回值
 
-表示对象的字符串。
+调用 `this.toString()` 的返回值。
 
 ## 描述
 
-{{jsxref("Object")}} `toLocaleString` 返回调用 {{jsxref("Object.toString", "toString()")}} 的结果。
+所有继承 `Object.prototype`（也就是说，除了 [`null`——原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象)）的对象都继承了 `toLocaleString()` 方法。{{jsxref("Object")}} 的 `toLocaleString` 返回调用 {{jsxref("Object/toString", "this.toString()")}} 的结果。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:28]:
```suggestion
所有继承 `Object.prototype`（也就是说，除了 [`null` 原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象)）的对象都继承了 `toLocaleString()` 方法。{{jsxref("Object")}} 的 `toLocaleString` 返回调用 {{jsxref("Object/toString", "this.toString()")}} 的结果。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 4]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+
+提供此方法是为了给对象一个通用的 `toLocaleString` 方法，即使不是所有对象都可以使用它。在核心语言中，这些内置对象重载 `toLocaleString` 以提供特定于语言环境的格式：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:30]:
```suggestion
提供此方法是为了给对象一个通用的 `toLocaleString` 方法，即使不是所有对象都可以使用它。在核心语言中，这些内置对象重写了 `toLocaleString` 以提供特定于语言环境的格式：
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 5]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+```js
+const obj = {
+  toString() {
+    return "My Object";
+  },
+};
+console.log(obj.toLocaleString()); // "My Object"
+```
+
+### 数组 toLocaleString() 的重载
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:53]:
```suggestion
### 数组重写的 toLocaleString()
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 6]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
 
-### 覆盖 `toLocaleString` 的对象
+const euroPrices = testArray.toLocaleString("fr", {
+  style: "currency",
+  currency: "EUR",
+});
+// "4,00 €,7,00 €,10,00 €"
+```
+
+### 日期对象 toLocaleString() 的重载
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:67]:
```suggestion
### 日期对象重写的 toLocaleString()
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 7]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+// "Fri May 29 2020 18:04:24 GMT+0100 (British Summer Time)"
+
+const deDate = testDate.toLocaleString("de");
+// "29.5.2020, 18:04:24"
+
+const frDate = testDate.toLocaleString("fr");
+// "29/05/2020, 18:04:24"
+```
+
+### 数字 toLocaleString() 的重载
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:82]:
```suggestion
### 数字重写的 toLocaleString()
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 8]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
-{{jsxref("Object")}} `toLocaleString` 返回调用 {{jsxref("Object.toString", "toString()")}} 的结果。
+所有继承 `Object.prototype`（也就是说，除了 [`null` 原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象)）的对象都继承了 `toLocaleString()` 方法。{{jsxref("Object")}} 的 `toLocaleString` 返回调用 {{jsxref("Object/toString", "this.toString()")}} 的结果。
+
+提供此方法是为了给对象一个通用的 `toLocaleString` 方法，即使不是所有对象都可以使用它。在核心语言中，这些内置对象重写了 `toLocaleString` 以提供特定于语言环境的格式：
+
+- {{jsxref("Array")}}: {{jsxref("Array.prototype.toLocaleString()")}}
+- {{jsxref("Number")}}: {{jsxref("Number.prototype.toLocaleString()")}}
+- {{jsxref("Date")}}: {{jsxref("Date.prototype.toLocaleString()")}}
+- {{jsxref("TypedArray")}}: {{jsxref("TypedArray.prototype.toLocaleString()")}}
+- {{jsxref("BigInt")}}: {{jsxref("BigInt.prototype.toLocaleString()")}}
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:36]:
```suggestion
- {{jsxref("Array")}}：{{jsxref("Array.prototype.toLocaleString()")}}
- {{jsxref("Number")}}：{{jsxref("Number.prototype.toLocaleString()")}}
- {{jsxref("Date")}}：{{jsxref("Date.prototype.toLocaleString()")}}
- {{jsxref("TypedArray")}}：{{jsxref("TypedArray.prototype.toLocaleString()")}}
- {{jsxref("BigInt")}}：{{jsxref("BigInt.prototype.toLocaleString()")}}
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 9]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+```js
+const obj = {
+  toString() {
+    return "My Object";
+  },
+};
+console.log(obj.toLocaleString()); // "My Object"
+```
+
+### 数组重写的 toLocaleString()
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:53]:
```suggestion
### Array 重写的 toLocaleString()
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 10]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+const testArray = [4, 7, 10];
+
+const euroPrices = testArray.toLocaleString("fr", {
+  style: "currency",
+  currency: "EUR",
+});
+// "4,00 €,7,00 €,10,00 €"
+```
+
+### 日期对象重写的 toLocaleString()
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:67]:
```suggestion
### Date 重写的 toLocaleString()
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 11]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
+// "Fri May 29 2020 18:04:24 GMT+0100 (British Summer Time)"
+
+const deDate = testDate.toLocaleString("de");
+// "29.5.2020, 18:04:24"
+
+const frDate = testDate.toLocaleString("fr");
+// "29/05/2020, 18:04:24"
+```
+
+### 数字重写的 toLocaleString()
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:82]:
```suggestion
### Number 重写的 toLocaleString()
```

**Review by jasonren0403 — COMMENTED** [review]
Array、Date、Number作为对象名称的话，或许可以不译，或许都写作“xx对象”

这里“数组”、“日期”还好，到“数字”就有些奇怪

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 12]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
 
 ## 描述
 
-{{jsxref("Object")}} `toLocaleString` 返回调用 {{jsxref("Object.toString", "toString()")}} 的结果。
-
-该函数提供给对象一个通用的`toLocaleString` 方法，即使不是全部都可以使用它。见下面的列表。
+所有继承 `Object.prototype`（也就是说，除了 [`null` 原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象)）的对象都继承了 `toLocaleString()` 方法。{{jsxref("Object")}} 的 `toLocaleString` 返回调用 {{jsxref("Object/toString", "this.toString()")}} 的结果。
 
-### 覆盖 `toLocaleString` 的对象
+提供此方法是为了给对象一个通用的 `toLocaleString` 方法，即使不是所有对象都可以使用它。在核心语言中，这些内置对象重写了 `toLocaleString` 以提供特定于语言环境的格式：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:30]: may not != could not

```suggestion
提供此方法是为了给对象一个通用的 `toLocaleString` 方法，即使不是所有对象都会使用它。在核心语言中，这些内置对象重写了 `toLocaleString` 以提供特定于语言环境的格式：
```

File: files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md [hunk 13]
```diff
@@ -3,31 +3,96 @@ title: Object.prototype.toLocaleString()
...
 - {{jsxref("Number")}}：{{jsxref("Number.prototype.toLocaleString()")}}
 - {{jsxref("Date")}}：{{jsxref("Date.prototype.toLocaleString()")}}
+- {{jsxref("TypedArray")}}：{{jsxref("TypedArray.prototype.toLocaleString()")}}
+- {{jsxref("BigInt")}}：{{jsxref("BigInt.prototype.toLocaleString()")}}
+
+## 示例
+
+### 使用基本的 toLocaleString() 方法
+
+基本的 `toLocaleString()` 方法只调用 `toString()`。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/tolocalestring/index.md:42]:
```suggestion
基本的 `toLocaleString()` 方法只是简单地调用 `toString()`。
```

**Review by yin1999 — APPROVED** [review]
Thanks @robert-jx

**Pull request merged**